src/requirements/requirement-graph-builder.ts

```diff
@@ -67,15 +67,17 @@ const buildRequirementFulfillmentGraph = <Requirement, Course, UserChoiceOnFulfi
   UserChoiceOnFulfillmentStrategy
 >): RequirementFulfillmentGraph<Requirement, Course> => {
   const graph = new RequirementFulfillmentGraph(getRequirementUniqueID, getCourseUniqueID);
-  const userCourseSet = new HashMap<Course, null>(getCourseUniqueID);
-  userCourses.forEach(course => userCourseSet.set(course, null));
+  const userCourseSet = new HashMap<Course, Course>(getCourseUniqueID);
+  userCourses.forEach(course => userCourseSet.set(course, course));
 
   // Phase 1:
   //   Building a rough graph by naively connecting requirements and courses based on
   //   `getAllCoursesThatCanPotentiallySatisfyRequirement`.
   requirements.forEach(requirement => {
+    graph.addRequirementNode(requirement);
```

Comment:
Without this, requirement without matching courses won't show up in the final array at all, which is bad.